### ISSUE TITLE
panic when set replica count and async replica

### ISSUE THREAD

**Issue by hulucc** [buraksezer/olric#93]
```
c.ReplicaCount = 3
c.ReplicationMode = config.AsyncReplicationMode
```
And start three instance, restart one of them, all panic
```
panic: runtime error: index out of range [-1]
spiracle-2 spiracle
spiracle-2 spiracle goroutine 2176 [running]:
spiracle-2 spiracle github.com/buraksezer/olric/internal/cluster/balancer.(*Balancer).backupCopies(0xc0004f2b90)
spiracle-2 spiracle     C:/Users/hulucc/go/pkg/mod/github.com/buraksezer/olric@v0.4.0-beta.2.0.20210502130407-be0cf149f919/internal/cluster/balancer/balancer.go:152 +0x69d
spiracle-2 spiracle github.com/buraksezer/olric/internal/cluster/balancer.(*Balancer).Balance(0xc0004f2b90)
spiracle-2 spiracle     C:/Users/hulucc/go/pkg/mod/github.com/buraksezer/olric@v0.4.0-beta.2.0.20210502130407-be0cf149f919/internal/cluster/balancer/balancer.go:193 +0xbb
spiracle-2 spiracle github.com/buraksezer/olric/internal/cluster/routingtable.(*RoutingTable).runCallbacks(0xc0006a0e00)
spiracle-2 spiracle     C:/Users/hulucc/go/pkg/mod/github.com/buraksezer/olric@v0.4.0-beta.2.0.20210502130407-be0cf149f919/internal/cluster/routingtable/callback.go:36 +0xb4
spiracle-2 spiracle created by github.com/buraksezer/olric/internal/cluster/routingtable.(*RoutingTable).updateRoutingOperation
spiracle-2 spiracle     C:/Users/hulucc/go/pkg/mod/github.com/buraksezer/olric@v0.4.0-beta.2.0.20210502130407-be0cf149f919/internal/cluster/routingtable/operations.go:122 +0x6b6
```

**Comment by buraksezer** [comment]
Hi,

I have implemented a fix to overcome the problem you reported but I need to test it extensively. I have also discovered a few other problems in `v0.4.0`. Some of them are severe problems and should be investigated carefully. 

By the way, `v0.4.0` has a different internal structure than the previous versions. Most of the clustering code has been rewritten and distributed data structures are refactored. All these changes should be tested extensively. I recommend using `v0.3.7`, if you don't need the improvements in `Stats` in `0.4.0` codebase.

**Comment by hulucc** [comment]
v0.3.7 has some problem on windows and v0.4.0 works fine.Also, I perfer the new version.

**Comment by buraksezer** [comment]
> v0.3.7 has some problem on windows and v0.4.0 works fine.Also, I perfer the new version.

I understand your consideration but the new version won't be ready for the next a couple of months. Could you please open an issue for the problem in `v0.3.7`? It may be easy to fix.

**Comment by buraksezer** [comment]
@hulucc I have just released a new version of [v0.4.x](https://github.com/buraksezer/olric/releases/tag/v0.4.0-beta.4). The problem should be disappeared by this update. I'm going to release a new version to fix #92. 

If you have any other problems, please don't hesitate to open an issue.
